### PR TITLE
Redesign Login and 2FA screen with TailwindCSS

### LINKED ISSUE #61: New Login Screen
Implement a new design for the login screen. I'm attaching the figma file to this issue, so anyone can open it and inspect the sizes, colors and etc. I'm trying to use components to organize the screens, making them more dynamic. As a suggestion, we can try to create the new screens more component-oriented, allowing us to easily customize them in the future.

---

## Login screen
Login-final

## Login screen with errors
Login with error

## Spacing information
login-spacings

---

# Grid

The screens were created using a 1280 canvas width, divided into 12 columns, with a 30px margin between them. You can trigger the grid on/off in figma.

image

# Fonts

**Font**: Lato (https://fonts.google.com/specimen/Lato)

# Components

## Heading 1
**Text size**: 32px
**Color**: #161616

## Form Input
**Text label size**: 16px
**Text label color**: #313131
**Input height**: 32px
**Input border**: 1px solid #C4C4C4
**Input background**: #F5F5F5

## Checkbox
**Size**: 20px X 20px
**Background color**: #F5F5F5

## Link
**Color**: #5285D4
**Size**: 12px

## Button
**Height**: 40px
**Text size**: 16px
**Background color**: #C51414

## Copyright information
**Text size**: 12px
**Text color**: #767676
**Line color**: #D6D6D6

# Figma File
[Punchclock.zip](https://github.com/Codeminer42/Punchclock/files/7478410/Punchclock.zip)

### PR DESCRIPTION
**Pull request by iagodahlem**
Fixes #61

### Screenshots

**Login**:

image

**Error**: 

image

### CODE REVIEW

**Comment by adbatista** [comment]
can we have screenshots?

**Comment by iagodahlem** [comment]
> can we have screenshots?

Done.

File: app/controllers/user/sessions_controller.rb [hunk 1]
```diff
@@ -8,8 +8,8 @@ class User::SessionsController < Devise::SessionsController
   def new
     super do |resource|
       if params[:commit].present?
-        resource.errors.add(:email, :invalid_authentication)
-        resource.errors.add(:password, :invalid_authentication)
+        resource.valid?
```
Comment (adbatista) [inline app/controllers/user/sessions_controller.rb:11]: you can use `resource.validate` here, is an alias for `valid?` but is more semantic in this case

File: app/controllers/user/sessions_controller.rb [hunk 2]
```diff
@@ -8,8 +8,8 @@ class User::SessionsController < Devise::SessionsController
...
+        resource.errors.add(:password, "não pode ficar em branco") if params[:password].blank?
```
Comment (adbatista) [inline app/controllers/user/sessions_controller.rb:12]: would be nice to have i18n here

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -3,19 +3,28 @@
   <head>
     <title>Punchclock</title>
     <meta name="viewport" content="width=device-width, initial-scale=1">
+
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+    <link href="https://fonts.googleapis.com/css2?family=Lato:ital,wght@0,100;0,300;0,400;0,700;1,100;1,300;1,400;1,700&display=swap" rel="stylesheet">
+
+    <%= stylesheet_link_tag "tailwind", "data-turbo-track": "reload" %>
```
Comment (adbatista) [inline app/views/layouts/application.html.erb:11]: there's no need to the data-turbo-track

Comment (adbatista) [inline app/views/layouts/application.html.erb:11]: why there changes if you have the redesign layout?

File: bin/dev [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+if ! command -v foreman &> /dev/null
+then
+  echo "Installing foreman..."
+  gem install foreman
+fi
+
+foreman start -f Procfile.dev
```
Comment (adbatista) [inline bin/dev:9]: why you had to add this file?

Comment (iagodahlem) [reply bin/dev:9]: It was added by the Tailwind gem generator, I thought it was a good addition to run the Procfile.dev easily. But we can remove you think it's not useful.

**Comment by adbatista** [comment]
update the screenshots pls

**Pull request merged**